lib/linter/linter.js

```diff
@@ -2291,10 +2153,9 @@ class Linter {
 		return applyDisableDirectives({
 			language: config.language,
 			sourceCode,
-			directives: commentDirectives.disableDirectives,
+			directives: commentDirectives,
 			disableFixes: options.disableFixes,
-			problems: lintingProblems
-				.concat(commentDirectives.problems)
+			problems: report.messages
 				.concat(inlineConfigProblems)
```

Comment:
`inlineConfigProblems` is effectively unused (it's always an empty array, as initialized), so we can remove it.